packages/components/src/components/radio/radio.test.stories.ts

```diff
@@ -0,0 +1,136 @@
+import '../../solid-components';
+import {
+  storybookDefaults,
+  storybookHelpers,
+  storybookTemplate,
+  storybookUtilities
+} from '../../../scripts/storybook/helper';
+
+const { argTypes, parameters } = storybookDefaults('sd-radio');
+const { generateTemplate } = storybookTemplate('sd-radio');
+const { overrideArgs } = storybookHelpers('sd-radio');
+const { generateScreenshotStory } = storybookUtilities;
+
+export default {
+  title: 'Components/sd-radio/Screenshot Tests',
+  tags: ['!autodocs'],
+  component: 'sd-radio',
+  args: overrideArgs([{ type: 'slot', name: 'default', value: 'Default Slot' }]),
+  argTypes,
+  parameters: { ...parameters }
+};
+
+/**
+ * Default: This shows sd-radio in its default state.
+ */
+
+export const Default = {
+  name: 'Default',
+  render: (args: any) => {
+    return generateTemplate({ args });
+  }
+};
+
+/**
+ * Use the size attribute to set the input radio desired size and the disabled attribute to disable the element. Clicks will be suppressed until the disabled state is removed
+ */
+
+export const DisabledAndSize = {
+  name: 'Disabled × Size',
+  parameters: { controls: { exclude: ['disabled', 'size', 'default'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        x: {
+          type: 'attribute',
+          name: 'disabled',
+          values: [false, true]
+        },
+        y: {
+          type: 'attribute',
+          name: 'size',
+          values: ['lg', 'sm']
+        }
+      },
+      constants: { type: 'attribute', name: 'disabled', value: true },
+      args
+    });
+  }
+};
+
+/**
+ * Use the `size` attribute to change the size of the input radio. This attribute affects the font-size within the element, while the element itself remains the same size.
+ */
+
+export const Size = {
+  name: 'Size',
+  parameters: { controls: { exclude: ['size'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        x: { type: 'attribute', name: 'size' }
+      },
+      args
+    });
+  }
+};
+
+/**
+ * Use the `invalid` attribute to mark the element is not in a format or a value the application will accept.
```

Comment:
Add:

Accessibility: For an invalid radio-group (since radios always come in groups) an error-text underneath the group is mandatory.